sdk/data/azappconfig/utils_test.go

```diff
@@ -0,0 +1,82 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azappconfig_test
+
+import (
+	"os"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azappconfig"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/recording"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	fakeConnStr = "Endpoint=https://contoso.azconfig.io;Id=fake-id:fake-value;Secret=ZmFrZS1zZWNyZXQ="
+)
+
+func TestMain(m *testing.M) {
+	err := recording.ResetProxy(nil)
+	if err != nil {
+		panic(err)
+	}
+
+	switch recording.GetRecordMode() {
+	case recording.PlaybackMode:
+		if err := recording.SetDefaultMatcher(nil, &recording.SetDefaultMatcherOptions{
+			ExcludedHeaders: []string{"Date", "X-Ms-Content-Sha256"},
+		}); err != nil {
+			panic(err)
+		}
+
+	case recording.RecordingMode:
+		defer func() {
```

Comment:
Doesn't `os.Exit` prevent this executing?

Reply:
Good catch!  fixed